Nodejs/Product/TestAdapter/TestFrameworks/NodejsTestInfo.cs

```diff
@@ -1,41 +1,62 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.IO;
+using System.Text;
 
 namespace Microsoft.NodejsTools.TestAdapter.TestFrameworks
 {
     internal class NodejsTestInfo
     {
-        public NodejsTestInfo(string fullyQualifiedName)
+        public NodejsTestInfo(string fullyQualifiedName, string modulePath)
         {
             var parts = fullyQualifiedName.Split(new[] { "::" }, StringSplitOptions.None);
             if (parts.Length != 3)
             {
                 throw new ArgumentException("Invalid fully qualified test name");
             }
-            ModulePath = parts[0];
-            TestName = parts[1];
-            TestFramework = parts[2];
+
+            this.ModulePath = modulePath;
+            this.ModuleName = parts[0];
+            this.TestName = parts[1];
+            this.TestFramework = parts[2];
         }
 
         public NodejsTestInfo(string modulePath, string testName, string testFramework, int line, int column)
         {
-            ModulePath = modulePath;
-            TestName = testName;
-            TestFramework = testFramework;
-            SourceLine = line;
-            SourceColumn = column;
+            var moduleName = Path.GetFileNameWithoutExtension(modulePath);
+            var hash = GetHash(modulePath);
+
+            this.ModulePath = modulePath;
+            this.ModuleName = $"{moduleName}[{hash}]";
+            this.TestName = testName;
+            this.TestFramework = testFramework;
+            this.SourceLine = line;
+            this.SourceColumn = column;
         }
 
-        public string FullyQualifiedName
+        private string GetHash(string filePath)
         {
-            get
+            try
+            {
+                using (var stream = File.OpenRead(filePath))
+                {
+                    var hash = Hash.GetFnvHashCode(stream);
+                    return Encoding.UTF8.GetString(hash);
+                }
+            }
+            catch (IOException)
             {
-                return ModulePath + "::" + TestName + "::" + TestFramework;
+                return "FILE_NOT_FOUND";
```

Comment:
I would return `null` here and have the caller `??` it with the error message.  Bonus points for using a localized error message.